### ISSUE TITLE
Extend cachedhcp to handle the proxydhcp packet

### ISSUE THREAD

**Issue by alkisg** [ipxe/ipxe#144]
Suppose that I boot from the PXE stack, which loads a custom build of undionly.kpxe with the following embedded ipxe script:

```shell
ifopen net0
chain ${proxydhcp/dhcp-server}/ltsp/ltsp.ipxe
```

The problem is that at that point, the proxydhcp variables aren't there; they're only available if a `dhcp net0` gets executed.

When using proxydhcp, there's little benefit in doing an additional DHCP request, as we have no control over the main DHCP server anyway, and our only variables that we'll need are ${proxydhcp/dhcp-server} and ${proxydhcp/filename}.

Would it be possible to get the ProxyDHCP packet and make it available at that point, to avoid the extra DHCP request?
Thank you again for the wonderful iPXE!

**Comment by Hammarskjold** [comment]
Talked to @mcb30 many times about this. Should be doable, just haven't been prioritized, but I agree it's a good thing as we can skip all extra DHCP requests when using a proxyDHCP. But for this to work also for EFI, if I remember correctly, the EFI subsystem doesn't cache the DHCP response at all. @mcb30 should we open up a separate issue on this?

**Comment by mcb30** [comment]
@Hammarskjold Yes, the UEFI cached DHCP support should probably be a separate issue.

**Comment by mcb30** [comment]
> @Hammarskjold Yes, the UEFI cached DHCP support should probably be a separate issue.

Created as #188

**Comment by mcb30** [comment]
Implemented for UEFI (but not yet legacy BIOS) in commit https://github.com/ipxe/ipxe/commit/e09e1142a

**Comment by alkisg** [comment]
Thanks for the UEFI implementation!
Any progress with the BIOS implementation?
I think it would also help in cases where the second DHCP fails (e.g. https://ipxe.org/err/4c1060 and cisco switches), and we resort to using pxelinux.0...

**Comment by NiKiZe** [comment]
Just noting state and references...
The logic implemented for EFI in https://github.com/ipxe/ipxe/blob/master/src/interface/efi/efi_cachedhcp.c#L82-L90
Is not yet done for pcbios https://github.com/ipxe/ipxe/blob/master/src/arch/x86/interface/pcbios/bios_cachedhcp.c
